.github/workflows/check_cla_assigned.yml

```
@@ -0,0 +1,174 @@
+name: Check CLA Signature for Issues and Pull Requests
+on:
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+  issues:
+    types:
+      - assigned
+permissions: {}
+jobs:
+  check-cla:
+    runs-on: ubuntu-24.04
+    permissions:
+      pull-requests: write
+      issues: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Install Dependencies
+        run: npm install googleapis google-auth-library
+      - name: Check CLA signature
+        id: check-cla
+        uses: actions/github-script@v6
+        env:
+          CLIENT_SECRET: '${{ secrets.SHEETS_CRED }}'
+          CLA_SHEET_ID: '${{ secrets.SPREADSHEET_ID }}'
+          USERNAME: '${{ github.event.pull_request.user.login || github.event.assignee.login }}'
+        with:
+          script: |
+            try {
+              const sheetsCredRaw = process.env.CLIENT_SECRET;
+              let sheetsCredJson;
+              sheetsCredJson = JSON.parse(sheetsCredRaw);
+              const username = process.env.USERNAME;
+              console.log(`Checking CLA signature for user: ${username}`);
+
+              const { google } = require('googleapis');
+              const { JWT } = require('google-auth-library');
+              const { OAuth2Client } = require('google-auth-library');
+
+              if (!sheetsCredJson.installed || !sheetsCredJson.installed.client_id || !sheetsCredJson.installed.client_secret || !sheetsCredJson.refresh_token) {
+                console.error("CLIENT_SECRET is missing required credential fields");
+                core.warning("CLIENT_SECRET is missing required credential fields");
+                core.setOutput('has-signed-cla', 'false');
```

this could still be confusing since we don't *know* that the CLA hasn't been signed. Instead, how about we don't set `has-signed-cla` at all and we change `if: steps.check-cla.outputs.has-signed-cla == 'false'` below to `if: steps.check-cla.outputs.has-signed-cla != 'true'` so that if anything goes wrong, we still fail closed?

---

Hmm, sounds good to me, and now I should raise error from `core.setFailed` and update below step condition with != true.